### PR TITLE
[Compute] vm boot-diagnostics enable: --storage becomes optional; New command: vm boot-diagnostics get-sas

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Resolve https://github.com/Azure/azure-cli/issues/14539

[Compute] vm boot-diagnostics enable: --storage becomes optional
[Compute] New command: vm boot-diagnostics get-sas

How to try it: https://github.com/Azure/azure-cli/blob/dev/doc/try_new_features_before_release.md

**Testing Guide**  

```
vm boot-diagnostics enable -g rg -n vm
az vm boot-diagnostics get-sas -g rg -n vm --expire 100
```

**History Notes**  

[Compute] vm boot-diagnostics enable: --storage becomes optional
[Compute] New command: vm boot-diagnostics get-boot-log-uris
[Compute] vm boot-diagnostics get-boot-log: support managed storage

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by arrownj — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -889,6 +889,18 @@
     crafted: true
 """
 
+helps['vm boot-diagnostics get-sas'] = """
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:892]: sas is [shared access signature](https://docs.microsoft.com/en-us/samples/azure-samples/storage-dotnet-sas-getting-started/storage-dotnet-sas-getting-started/), while the [request](https://github.com/Azure/azure-cli/issues/14539) is asking to get boot log url. Suggest to rename to `get-log`

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py:892]: A sample response:
```
{
  "consoleScreenshotBlobUri": "https://md-visib0spjzkmsn3z.blob.core.windows.net:443/3289a9bc-c41f-4750-9595-a6a41500fc30/vm1.66212157-7e08-48e9-bb0b-1b31fad6ea79.screenshot.bmp?sv=2017-04-17&sr=b&sig=z5YDfqrK93Lhd8QzWzTeRquDFVLalKmbkt5nZWppIk8%3d&se=2020-09-14T08%3a03%3a52Z&sp=r",
  "serialConsoleLogBlobUri": "https://md-visib0spjzkmsn3z.blob.core.windows.net:443/3289a9bc-c41f-4750-9595-a6a41500fc30/vm1.66212157-7e08-48e9-bb0b-1b31fad6ea79.serialconsole.log?sv=2017-04-17&sr=b&sig=aaQNEtL2D%2bBgT1VtLjFkMx%2buaNRrMxM3CrTcXxwyvUs%3d&se=2020-09-14T08%3a03%3a52Z&sp=r"
}
```
how about get-blog-uri?

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 2]
```diff
@@ -863,7 +863,7 @@
 short-summary: Enable the boot diagnostics on a VM.
 parameters:
   - name: --storage
-    short-summary: Name or URI of a storage account (e.g. https://your_storage_account_name.blob.core.windows.net/)
+    short-summary: Name or URI of a storage account (e.g. https://your_storage_account_name.blob.core.windows.net/). If not specified, managed storage is used.
```
Comment (bim-msft) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:866]:
```suggestion
    short-summary: Name or URI of a storage account (e.g. https://your_storage_account_name.blob.core.windows.net/). If it's not specified, managed storage will be used.
```

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py:866]: A little contribution to grammar.

**Review by bim-msft — CHANGES_REQUESTED** [review]

**Review by bim-msft — APPROVED** [review]

**Review by haroldrandom — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/_help.py (hunk 1 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py:892]: thanks @qwordy . As we synced offline, with new managed storage  for boot diagnostic log,  2 changes need.
-  **MUST** change existing get_boot_log command to support showing content from managed storage uri, when log is managed storage, get uri from the new InstanceView API
- **Nice to have** add a new command `get_boot_log_uris` to show boot log uris. This new command need to cover both scenarios also: managed storage, and user specified storage.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py]: Done

**Pull request merged**